src/test/java/hudson/plugins/ec2/util/AmazonEC2FactoryExtensionFilter.java

```diff
@@ -0,0 +1,20 @@
+package hudson.plugins.ec2.util;
+
+import hudson.Extension;
+import hudson.ExtensionComponent;
+import jenkins.ExtensionFilter;
+
+/**
+ * The sole purpose of this class is to filter out {@link AmazonEC2FactoryImpl} so as to avoid doing real calls to
+ * EC2 when running tests.
+ */
+@Extension
```

Comment:
```suggestion
@TestExtension
```

Reply:
I just tried `@TestExtension` out, but Jenkins won't pick it up then.

https://wiki.jenkins.io/display/JENKINS/Unit+Test

seems to suggest that `@TestExtension` should be used as an inner class for a specific unit test. I.e. I think each test class that needs them would need their own inner class implementation for that to work.

Reply:
Oh, I did not realize that. Ah well. An alternative would be yet another annotation, but I think that's just adding unnecessary code at that point. You can leave this as `@Extension`.